src/Http/Paginators/PageRequestPaginator.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Http\Paginators;
+
+use Saloon\Contracts\Connector;
+use Saloon\Contracts\Request;
+use Saloon\Traits\Request\HasPagedPagination;
+
+// TODO 1: Look into serialising the Connector and original Request,
+//           to ensure that we can rebuild the paginator state without storing the entire multiverse.
+
+class PageRequestPaginator extends RequestPaginator
+{
+    use HasPagedPagination;
+
+    /**
+     * @var int
+     */
+    protected readonly int $originalPage;
+
+    /**
+     * @param \Saloon\Contracts\Connector $connector
+     * @param \Saloon\Contracts\Request $originalRequest
+     * @param int|null $limit
+     * @param int $page
+     */
+    public function __construct(
+        Connector $connector,
+        Request $originalRequest,
+        ?int $limit = null,
+        int $page = 1,
+    ) {
+        parent::__construct($connector, $originalRequest, $limit);
+
+        $this->currentPage = $this->originalPage = $page;
+    }
+
+    /**
+     * @return int
+     */
+    public function totalEntries(): int
+    {
+        // Make sure we have a response.
+        if (is_null($this->currentResponse)) {
+            $this->current();
+        }
+
+        return $this->currentResponse->json('json');
```

Comment:
Are we able to still allow people to overwrite these? We could also provide another variable so people can find the "total" in requests e.g

```php
return $this->currentResponse->json($this->getTotalKey());
```